tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -810,3 +810,59 @@ def solve_tridiag(
     solve_tridiag(a, b, c, d, out=out, offset_provider={})
 
     np.allclose(expected, out)
+
+
+def test_ternary_operator(reduction_setup):
+    rs = reduction_setup
+    Edge = rs.Edge
+
+    num_edges = rs.num_edges
+    a = np_as_located_field(Edge)(np.ones((num_edges,)))
+    b = np_as_located_field(Edge)(2 * np.ones((num_edges,)))
+    out = np_as_located_field(Edge)(np.zeros((num_edges,)))
+
+    @field_operator
+    def ternary_field_op(a: Field[[Edge], float], b: Field[[Edge], float]) -> Field[[Edge], float]:
+        c = a if 2 < 3 else b
+        return c
+
+    @program
+    def ternary_field(a: Field[[Edge], float], b: Field[[Edge], float], out: Field[[Edge], float]):
+        ternary_field_op(a, b, out=out)
+
+    ternary_field(a, b, out, offset_provider={})
+    e = np.asarray(a) if 2 < 3 else np.asarray(b)
+    np.allclose(e, out)
+
+
+def test_ternary_operator_tuple(reduction_setup):
+    rs = reduction_setup
+    Edge = rs.Edge
+
+    num_edges = rs.num_edges
```

Comment:
```suggestion
    Edge = reduction_setup.Edge
    num_edges = reduction_setup.num_edges
```
You are mixing different styles here. Either use `rs = reduction_setup` and then `rs.what_you_need` for extract what you need, but skip the `rs` aliasing.